lib/instrumentation/modules/http.js

```diff
@@ -41,6 +41,11 @@ module.exports = function (http, agent, { enabled }) {
         // It shouldn't be possible for the statusCode to be falsy, but just in
         // case we're in a bad state we should avoid throwing
         trans.result = 'HTTP ' + (this.statusCode || '').toString()[0] + 'xx'
+        if (this.statusCode >= 500) {
+          trans.outcome = 'failure'
+        } else {
+          trans.outcome = 'success'
+        }
 
```

Comment:
Here I see that during our handling/wrapping of the `http.ServerResponse` method we're examining the status code used in the response, and if it's greater than or equal to 500 we set the outcome as a failure -- otherwise we set the outcome to success.  This matches the behavior described in the spec.